### PR TITLE
Fix import warning

### LINKED ISSUE #477: Unable to build mlx_lm from source
After successfully building mlx (on an Apple Mac Studio Z14K0007J M1 Ultra) from git main, I tried to building mlx_lm the way I usually do from my local copy of mlx-examples:

```bash
% pip install -U -e llms
Obtaining file:///Users/oori/medical_llm/mlx-examples/llms
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      Traceback (most recent call last):
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 132, in get_requires_for_build_editable
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 441, in get_requires_for_build_editable
          return self.get_requires_for_build_wheel(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
        File "/Users/oori/medical_llm/mlx-examples/llms/mlx_lm/__init__.py", line 1, in <module>
          from .convert import convert
        File "/Users/oori/medical_llm/mlx-examples/llms/mlx_lm/convert.py", line 9, in <module>
          import mlx.core as mx
      ModuleNotFoundError: No module named 'mlx'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### PR DESCRIPTION
**Pull request by awni**
Refactor `convert` API into new file to avoid import warning when using it from the CLI:

```
/Users/awnihannun/miniconda3/lib/python3.9/runpy.py:127: RuntimeWarning: 'mlx_lm.convert' found in sys.modules after import of package 'mlx_lm', but prior to execution of 'mlx_lm.convert'; this may result in unpredictable behaviour
```

Also closes #477 (build breaks trying to import mlx_lm when building)

### CODE REVIEW

File: llms/mlx_lm/api.py
Comment (mzbac) [inline llms/mlx_lm/api.py:1]: Just curious, will `api.py` be the place for model conversion related utilities? The name of `api.py` gives some impression of an API server.

Comment (awni) [reply llms/mlx_lm/api.py:1]: Ehh it's a bit muddled.

Basically the issue is from the fact that we want both a CLI and an API (e.g. `pyhon -m mlx_lm.convert` and `from mlx_lm import convert`). Any time you have that you have to have the function you are importing in `__init__.py` live outside the CLI `*.py` file.

So for example if we wanted to add `from mlx_lm import merge` we should move the main `merge` function into `api.py`.

I'm definitely all ears for better naming suggestions. But I think it's good to rid that warning which sounds a bit scary.

Comment (mzbac) [reply llms/mlx_lm/api.py:1]: Yeah, makes sense. As for naming... I don't want to get into the hardest thing in programming :p

Comment (awni) [reply llms/mlx_lm/api.py:1]: What if I just put it in utils for now and remove api.py? Preferable?

Comment (mzbac) [reply llms/mlx_lm/api.py:1]: Utils is a bit too large and hard to maintain now. Does "cli.py" make sense?

Comment (awni) [reply llms/mlx_lm/api.py:1]: Another alternative might be to move all the CLI commands into `cli.py` and then provide the appropriate mappings in `setup.py`. I'm not sure if that will work, but I can look at it.

**Comment by awni** [comment]
I moved everything to utils.py for now. It seems like a good enough place given that's where `generate` already was.

One option I considered is to do a [console_script with setup.py](https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html) but that would be a larger diff / change to the CLI, so maybe makes more sense as a follow up. It would be nice though because:

- It would simplify the CLI from `python -m mlx_lm ...` to `mlx_lm .. ` (or whatever name you want)
- It would make it easy to separate the CLI commands into their own file

**Comment by awni** [comment]
@angeloskath this is good to go. Just fixes a couple bugs with building / using MLX LM.

**Comment by mzbac** [comment]
> console_script with setup.py

This would be a great option and could reduce confusion between the package name as `mlx-lm`, but when using it, the user has to type `mlx_lm.generate`. I can take a look at how to do this and maybe create a refactor PR later for this.

**Review by mzbac — APPROVED** [review]

**Review by angeloskath — APPROVED** [review]

**Pull request merged**